apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -92,16 +159,35 @@ export const assignments = state =>
  */
 export const currentAssignmentIndex = (state, sectionId) => {
   const section = state.sections[sectionId];
-  const assignmentList = assignments(state);
-  const target = {
-    courseId: section.courseId,
-    scriptId: section.courseId ? null : section.scriptId
-  };
 
-  // Find an assignment with the appropriate id
-  const assignmentIndex = assignmentList.findIndex(assignment => (
-    assignment.courseId === target.courseId && assignment.scriptId === target.scriptId
-  ));
+  const assignmentIndex = getAssignmentIndex(assignments(state))(
+    section.courseId, section.scriptId);
 
   return assignmentIndex === -1 ? null : assignmentIndex;
 };
+
+/**
+ * Find an assignment with the appropriate id. If both courseId and scriptId are
+ * non-null, look only at the courseId.
+ * This is memoized at two separate levels because JS doesnt have any good way
+ * to have a cache key consisting of an object (assignmentList) and two strings.
+ * @param {object[]} assignmentList - array of valid courses/scripts
+ * @param {string} courseId - course id of the assignment we're looking for
+ * @param {string} scriptId - script id of the assignment we're looking for
+ */
+const getAssignmentIndex = _.memoize(assignmentList => (
+  _.memoize((courseId, scriptId) => (
+    assignmentList.findIndex(assignment =>
+      assignment.courseId === courseId &&
+      (courseId || assignment.scriptId === scriptId)
+    )
+  ), (courseId, scriptId) => `${courseId}_${scriptId}`)
+));
```

Comment:
:confused: I didn't really follow that, but I know I don't like `getAssignmentIndex(assignments(state))(section.courseId, section.scriptId)`.